tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,103 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeDirectParent()
```

Comment:
This one use the store so it's too specific to this repo IMO. But I agree we should test this feature the same way everywhere, there is a bit of work here.